packages/headless/src/features/did-you-mean/did-you-mean-actions.ts

```diff
@@ -21,3 +21,11 @@ export const applyDidYouMeanCorrection = createAction(
   'didYouMean/correction',
   (payload: string) => validatePayload(payload, requiredNonEmptyString)
 );
+
+/**
+ * Set the original query performed when a query correction is applied
+ */
+export const didYouMeanCorrectionReceived = createAction(
+  'didYouMean/correctionReceived',
+  (payload: string) => validatePayload(payload, requiredNonEmptyString)
```

Comment:
**Preference**

Would have been possible to leverage the query on the `executeSearch.fulfilled` payload, avoiding the need to introduce a new action?

Isn't the "DYM correction received" the same as the `if` branch in the [reducer](https://github.com/coveo/ui-kit/pull/675/files#diff-44ef2baf1a421be26fc2a784913863a2e947bae5d7c5d41aafb313dae804f29cR27)? 

If we can embed the logic inside the reducer, this would be ideal.


**On the other hand**

If this action is needed, then I would prefer to use a more descriptive argument name than `payload` so that a user of the action would know what string to pass. e.g. `originalQuery`.

Reply:
Seconded, the query already goes through the state at some point, it can be retrieved

Reply:
The only problem is that in executeSearch.fulfilled, the `executed query` we receive is the corrected one, not the original one.

If you prefer, I can add a `originalQuery` for executeSearch payload, which would be identical to `queryExecuted` 99% of the time, but when there's a correction it would be different.

I could also just augment/tweak the `queryCorrection` array that comes from the API to embed the original complete query.

I'm going to go with the latter.

Reply:
If it is embedded in the array, how would one know what is original and what is a correction without inspecting the values at run-time? I feel your first suggestion of adding a named key in an object would be more explicit?

```js
{
    originalQuery: ...,
    queryExecuted: ...
}
```